homeassistant/components/media_player/squeezebox.py

```diff
@@ -0,0 +1,277 @@
+"""
+homeassistant.components.media_player.squeezebox
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Provides an interface to the Logitech SqueezeBox API
+
+Configuration:
+
+To use SqueezeBox add something like this to your configuration:
+
+media_player:
+  platform: squeezebox
+  name: SqueezeBox
+  server: 192.168.1.21
+  player: Player1
+  port: 9090
+  user: user
+  password: password
+
+Variables:
+
+name
+*Optional
+The name of the device
+
+server
+*Required
+The address of the Logitech Media Server
+
+player
+*Required
+The unique name of the player
+
+port
+*Optional
+Telnet port to Logitech Media Server, default 9090
+
+user
+*Optional
+User, if password protection is enabled
+
+password
+*Optional
+Password, if password protection is enabled
+"""
+
+import logging
+import telnetlib
+import urllib.parse
+
+from homeassistant.components.media_player import (
+    MediaPlayerDevice, SUPPORT_PAUSE, SUPPORT_SEEK, SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_MUTE, SUPPORT_PREVIOUS_TRACK, SUPPORT_NEXT_TRACK,
+    MEDIA_TYPE_MUSIC)
+from homeassistant.const import (
+    STATE_IDLE, STATE_PLAYING, STATE_PAUSED, STATE_OFF, STATE_UNKNOWN)
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_SQUEEZEBOX = SUPPORT_PAUSE | SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE |\
+    SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | SUPPORT_SEEK
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the squeezebox platform. """
+    add_devices([
+        SqueezeBoxDevice(
+            config.get('name', 'SqueezeBox'),
+            config.get('server'),
```

Comment:
It seems like I can't use homeassistant.helpers.validate_config since a media_player device does not get the full config passed to setup_platform, only the subset under the media_player domain. 

I'll write my own validation